### PR TITLE
RavenDB-11928 Auto assign backup task name when missing during update

### PR DESCRIPTION
**Pull request by ml054**

### CODE REVIEW

File: src/Raven.Server/ServerWide/Commands/PeriodicBackup/UpdatePeriodicBackupCommand.cs [hunk 1]
```diff
@@ -26,13 +26,14 @@ public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
...
                 Configuration.TaskId = etag;
-                if (string.IsNullOrEmpty(Configuration.Name))
-                {
-                    Configuration.Name = record.EnsureUniqueTaskName(Configuration.GetDefaultTaskName());
-                }
-
-                record.EnsureTaskNameIsNotUsed(Configuration.Name);
             }
+            
+            if (string.IsNullOrEmpty(Configuration.Name))
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/Commands/PeriodicBackup/UpdatePeriodicBackupCommand.cs]: @ayende if task had a name (`TaskId != 0`), do we want to regenerate it if on update `null` will be send or throw?

Comment (ayende) [reply src/Raven.Server/ServerWide/Commands/PeriodicBackup/UpdatePeriodicBackupCommand.cs]: The question here is, why not use the _old_ name?

Comment (ppekrol) [reply src/Raven.Server/ServerWide/Commands/PeriodicBackup/UpdatePeriodicBackupCommand.cs]: We can do that if 'name' is null. @ml054 can you apply that?

Comment (ml054) [reply src/Raven.Server/ServerWide/Commands/PeriodicBackup/UpdatePeriodicBackupCommand.cs]: I've applied that same logic we had here as in UpdateExternalReplicationCommand

**Pull request merged**